### ISSUE TITLE
ERROR: Unhandled exception: <b>UnboundLocalError</b>:cannot access local variable 'src' where it is not associated with a value

### ISSUE THREAD

**Issue by xxpowerxx1qz** [JimmXinu/EpubSplit#15]
calibre, version 7.21.0

ERROR: Unhandled exception: <b>UnboundLocalError</b>:cannot access local variable 'src' where it is not associated with a value

calibre 7.21  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 17, 0) && Package KFX (from KFX Input) (2, 17, 0) && EpubMerge (3, 2, 0) && EpubSplit (3, 9, 0) && FanFicFare (4, 40, 0) && KFX metadata reader (from KFX Input) (2, 17, 0) && From KFX (2, 17, 0) && KFX Input (2, 17, 0) && KindleUnpack - The Plugin (0, 83, 1)
Traceback (most recent call last):
  File "calibre_plugins.epubsplit.dialogs", line 382, in do_loop
  File "calibre_plugins.epubsplit.epubsplit_plugin", line 261, in _do_splits_loop
  File "calibre_plugins.epubsplit.epubsplit_plugin", line 456, in _do_split
  File "calibre_plugins.epubsplit.epubsplit", line 873, in write_split_epub
  File "calibre_plugins.epubsplit.epubsplit", line 835, in get_split_files
  File "calibre_plugins.epubsplit.epubsplit", line 1174, in add_content_file
UnboundLocalError: cannot access local variable 'src' where it is not associated with a value

Was hoping I can get some kind of assistance with this. Trying to split apart a rather large epub anthology I created with FanFicFare. Was nearly 3gb before I compressed the images to bring the overall size down to 900mb but my mobile ereaders are still struggling to open it so I thought axing it into even smaller parts would help but I keep running into this error which seems to halt any progress in the splitting process.

**Comment by JimmXinu** [comment]
That error will only happen when there's an `<img>` tag that doesn't have a `src` or `xlink:href` attribute pointing to its source file.

I can give you a version of epubmerge that corrects that, but if there are any other issues (or there's another attribute to look at), it would be quicker to fix if you could give me a copy of your epub to test.

A link to a share you can then remove would be fine--vs posting it here where github keeps everything, forever.

**Comment by xxpowerxx1qz** [comment]
[https://www.dropbox.com/scl/fo/2wwhd8y1s573j724w6urt/AEj7nRr3GpDiU7rQ5YKRVg8?rlkey=z66436usglvi4tv5dlydmw8gq&st=bzeey45a&dl=0](url)

Here you are. Lemme know if it works as I've never shared a file on Dropbox before. Another version of epubmerge wouldn't help me since this file took 28 hours to finish and I'm out of the country, remoting into my PC to access Calibre.

My thanks and if you can help me figure out how to split this into 3 or 4 pieces, it would be awesome. I appreciate all the work you do. FanFicFare is an amazing and feature-full piece of software. I was expecting to spend an few hours slowly feeding it SpaceBattles links to parse but than I saw it could parse multiple URLs so I dumped in my OneTab page and it found all 4k+ links. I cut it down to the recent 1200 ish cause the original 4k would probably have taken a literal week to fetch and download, but your work is deeply appreciated.

**Comment by JimmXinu** [comment]
Downloaded and looking at it now--I expect I'll have a fixed test version up later today.

Glad you like FFF.  But be aware that many (or lengthy) downloads can get you blocked by the site.  

1208 stories at once is 50-100 times more than I recommend downloading at one time as a general rule.

I would also recommend downloading all those (apparently unrelated?) stories as individual epubs.  It's safer (only one story would affected by a failure instead of all of them), and as you've found, many tools don't like files of that size.  

For example, I had time to write all this up while waiting for the file to load...

**Comment by xxpowerxx1qz** [comment]
Yeah, I procrastinated downloading the stories so I basically went for broke. I was honestly expecting the file to have failed and Calibre crashing and burning when I remote'd in but was pleasantly surprised when it seemed to work. But yeah, touching the file is rough. Doing any big thing to the file basically causes Calibre to stop responding but it recovers _eventually._ I just log off and come back later. It was why I was trying to chop it down to make it easier to handle, and hopefully, actually usable by mobile epub reading apps. They just kinda die as it is.

**Comment by JimmXinu** [comment]
As I thought, this error came from `<img>` tags without image source attributes.  Or at least, that's the only error I saw before it failed.

I can't recommend using EpubSplit with this particular epub.  It ran for more than an hour trying on my computer and finally failed with a memory error.  This file is just too big for EM to work with.

On the other hand, FanFicFare uses the EpubMerge plugin to create anthologies.  You should be able to use EpubMerge's UnMerge option to split this anthology epub into individual story epubs in Calibre without needing to download them again.  Quicker than trying to split, but it still took ~20 minutes for me.

**Comment by JimmXinu** [comment]
Test version posted [here](https://www.mobileread.com/forums/showthread.php?p=4469346#post4469346).
